pkg/network/ebpf/c/shared-libraries/probes.h

```diff
@@ -82,19 +82,19 @@ static __always_inline void do_sys_open_helper_exit(exit_sys_ctx *args) {
         goto cleanup;
     }
 
-    if (!match6chars(0, 'l','i','b','s','s','l') && !match6chars(0, 'c','r','y','p','t','o') && !match6chars(0, 'g','n','u','t','l','s')) {
+    if (!match6chars(0, 'l', 'i', 'b', 's', 's', 'l') && !match6chars(0, 'c', 'r', 'y', 'p', 't', 'o') && !match6chars(0, 'g', 'n', 'u', 't', 'l', 's') && !match6chars(0, 'c', 'u', 'd', 'a', 'r', 't')) {
```

Comment:
I'm concerned about the change
did you very it doe snot impact USM performance?

We added this condition to avoid bombarding the usermode with redundant events.
I'm more concerned about the future rather than this change, as more libraries will be added, it can impact the performance of USM but also, the complexity and size of our probes.

Reply:
The impact depends on how many processes are opening the `libcudart` library. Taking into account that it's not a library shipped by default and that processes that use it tend to be long running, in most deployments we probably wouldn't see any significant increase in usermode events.

I have thought about this problem too. I think with the new uprobe-attacher it will be easier to tackle, my idea was to define different programs for different library sets (e.g., one for crypto libs, another for GPU libs, etc) and make the attacher choose the correct one based on the library configured by the user.

Reply:
By the way, I wasn't planning to merge this before the freeze, so we should have far more time in staging to ensure there's no issues with a variety of loads.

Reply:
> The impact depends on how many processes are opening the libcudart library. Taking into account that it's not a library shipped by default and that processes that use it tend to be long running, in most deployments we probably wouldn't see any significant increase in usermode events.

That's one type of impact.
The other one is the impact on the caller (application which opened a file), and we're adding more logic to our probe
Other impacts on the USM probes are increased instruction count and complexity

> I have thought about this problem too. I think with the new uprobe-attacher it will be easier to tackle, my idea was to define different programs for different library sets (e.g., one for crypto libs, another for GPU libs, etc) and make the attacher choose the correct one based on the library configured by the user.

That sounds good to me